.github/workflows/run-checks.yml

```diff
@@ -0,0 +1,88 @@
+name: "Run Checks"
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+permissions:
+  contents: read
+  pull-requests: write
+
+jobs:
+  check-control-station:
+    name: "Check Control Station"
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: ./control-station
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v4
+
+      - name: Set up Node
+        uses: actions/setup-node@v4
+        with:
+          node-version: 20
+
+      - name: Install frontend dependencies
+        run: npm install
+
+      - name: Format with Prettier
+        uses: EPMatt/reviewdog-action-prettier@v1
+        with:
+          workdir: control-station/
+          level: warning
+          reporter: github-pr-review
+          prettier_flags: src
+
+      - name: Lint with ESLint (review)
+        uses: reviewdog/action-eslint@v1
+        if: github.event_name == 'pull_request'
+        with:
+          workdir: control-station/
+          level: error
+          reporter: github-pr-review
+          eslint_flags: src
+
+      - name: Lint with ESLint (check)
+        uses: reviewdog/action-eslint@v1
+        if: github.event_name == 'push'
+        with:
+          workDir: control-station/
```

Comment:
Chore: the `workdir` input should be lowercase as described in [reviewdog/action-eslint](https://github.com/reviewdog/action-eslint) although I'm not sure if it actually matters (seemed to work before otherwise).